language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -415,45 +491,106 @@ impl DependencyGraph {
             bail!("Merging dependencies into a graph after calculating its 'always' dependencies");
         }
 
-        for (ext_name, mut ext_pkg) in ext_table {
-            ext_pkg.resolver = Some(resolver);
+        if ext_table.is_empty() {
+            // the external graph is effectively empty - nothing to merge
+            return Ok(());
+        }
 
-            // The root package is not present in the package table (because it doesn't have a
-            // source).  If it appears in the other table, it indicates a cycle.
-            if ext_name == self.root_package {
-                bail!(
-                    "Conflicting dependencies found:\n{0} = 'root'\n{0} = {1}",
-                    ext_name,
-                    PackageWithResolverTOML(&ext_pkg),
-                );
-            }
+        // unwrap safe as the table must have the package if the graph has it
+        let merged_pkg = ext_table.get(&merged_pkg_name).unwrap();
+        // unwrap is safe as all edges have a Dependency weight
+        let merged_dep = ext_graph.edge_weight(from, merged_pkg_name).unwrap();
+        self.package_graph
+            .add_edge(from, merged_pkg_name, merged_dep.clone());
```

Comment:
nit: Is it possible to merge the package first, and then add the edge if that succeeds?

Reply:
Should be!